### PR TITLE
cmd/gobgp: use sd_notify protocol to notify the status change to systemd

### PR DESCRIPTION
**Pull request by Koaxiel**
It allows to notify the start-up completion of `gobgpd` to [systemd](https://github.com/systemd) via [sd_notify](https://www.freedesktop.org/software/systemd/man/sd_notify.html).

### CODE REVIEW

File: cmd/gobgpd/main.go [hunk 1]
```diff
@@ -205,6 +206,14 @@ func main() {
 	bgpServer := server.NewBgpServer(server.GrpcListenAddress(opts.GrpcHosts), server.GrpcOption(grpcOpts))
 	go bgpServer.Serve()
 
+	if status, err := daemon.SdNotify(false, daemon.SdNotifyReady); !status {
+		if err != nil {
```
Comment (fujita) [inline cmd/gobgpd/main.go]: If systemd is not available on a system, we get a warning. I don't think that it's reasonable. Can you add an option to enable this feature?

File: cmd/gobgpd/main.go [hunk 2]
```diff
@@ -26,6 +26,7 @@ import (
 	"runtime"
 	"syscall"
 
+	"github.com/coreos/go-systemd/daemon"
```
Comment (fujita) [inline cmd/gobgpd/main.go]: Needs to update the dependency

Comment (Koaxiel) [reply cmd/gobgpd/main.go]: Done

File: cmd/gobgpd/main.go (hunk 1 above)
Comment (Koaxiel) [reply cmd/gobgpd/main.go]: I added an option and updated the example of the systemd service

**Comment by fujita** [comment]
Also needs to update files for the module but I'll take care of it. Thanks a lot!

**Pull request closed without merging**